GVFS/GVFS.UnitTests/Common/NuGetUpgrade/NuGetUpgraderTests.cs

```diff
@@ -23,7 +25,7 @@ public class NuGetUpgraderTests
         private const string NewerVersion = "1.6.1185.0";
         private const string NewerVersion2 = "1.7.1185.0";
 
-        private const string NuGetFeedUrl = "feedUrlValue";
+        private const string NuGetFeedUrl = "https://pkgs.dev.azure.com/contoso/packages";

```

Comment:
This looks like a real URL, is there a fake one we can use to ensure our tests don't accidentally hit the real one? (e.g. something like "mock://pkgs.dev.azure.com/vfs4gunittest/packages")

Reply:
I like the suggestion - one wrinkle is that the code being testing is looking for a real URL (that starts with `https://`). I could tweak the URL parsing code to accept `mock://` as well, but I didn't want to make the product code change just to support using `mock://` URLs in test code. Maybe there is another way to handle this... Thoughts?

Reply:
> one wrinkle is that the code being testing is looking for a real URL (that starts with https://).

Ahh, good point.

> Maybe there is another way to handle this... Thoughts?

We could include characters that are invalid (e.g. it sounds like [TAB is invalid](https://stackoverflow.com/questions/1547899/which-characters-make-a-url-invalid)), but I'm not sure if that would actually stop clients from sending a request to `https://pkgs.dev.azure.com`.

Maybe there's not much more we can do here?  (But if contoso is a real package feed, perhaps we should switch to one that belongs to VFS4G instead?)

Reply:
In looking at the code seems like we need a `NuGetFeedUrlToAzDevOrgUrlConverter` that has an interface so that we can have other implementations like a `NuGetFeedUrlToGitHubUrlConverter`, etc?  Then we could have one to pass in for tests that verifies `mock://` instead of `https://`?  No idea how much work or how disruptive this would be though.

Reply:
I like this idea, but was going to defer actually making it